### PR TITLE
Clarify the reason why there is the restriction on changing the set of stores

### PR DESCRIPTION
**Pull request by vkrol**
https://github.com/mobxjs/mobx-react/issues/745#issuecomment-520255411

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -308,10 +308,13 @@ Notes:
...
 -   When using both `@inject` and `@observer`, make sure to apply them in the correct order: `observer` should be the inner decorator, `inject` the outer. There might be additional decorators in between.
 -   The original component wrapped by `inject` is available as the `wrappedComponent` property of the created higher order component.
 
+#### "The set of provided stores has changed" error
+Values provided through `Provider` should be final. Make sure that if you put things in `context` that might change over time, that they are `@observable` or provide some other means to listen to changes, like callbacks. However, if your stores will change over time, like an observable value of another store, MobX will throw an error.
+This restriction exists mainly for legacy reasons. If you have a scenario where you need to modify the set of stores, please leave a comment about it in this issue https://github.com/mobxjs/mobx-react/issues/745.
```
Comment (FredyC) [inline README.md]:
```suggestion
This restriction exists mainly for legacy reasons. If you have a scenario where you need to modify the set of stores, please leave a comment about it in this issue https://github.com/mobxjs/mobx-react/issues/745. Or a preferred way is to [use React Context](https://mobx-react.js.org/recipes-context) directly which does not have this restriction.
```

**Comment by FredyC** [comment]
Thanks

**Comment by nebosite** [comment]
@FredyC , since the thread [here](https://github.com/mobxjs/mobx-react#the-set-of-provided-stores-has-changed-error) is locked, I'll post my comment here:

I ran into the "provided stores has changed" exception today trying to do what felt pretty innocent based on my data-binding experience with XAML apps.  Basically, I'm showing a set of controls based on an array of models.  The onClick of the refresh button tries to swap out one of the models in the client array for a new model.  This is where the exception happens.  

Here is the UI code:

```
    render() {
        const {gameTestModel} = this.props;
        if(gameTestModel.presenterSize === 0) this.updatePresenterSize(window.innerWidth);
        
        const renderClient = (clientNumber: number) => {
            const clientModel = gameTestModel.clientModels[clientNumber];

            return (
                <React.Fragment>
                    <button onClick={()=>gameTestModel.refreshClientModel(clientNumber)}>Refresh</button>
                    <LobbyMainPage lobbyModel={clientModel} /> 
                </React.Fragment>)
        }

        return (
            <div className="gameTestComponent">
                <table>
                    <tbody>
                        <tr>
                            { [0,1,2,3].map(n => <td key={n}>{renderClient(n)}</td> )}
                        </tr>
                    </tbody>
                </table>
            </div>
        );
    }
```

Here is the relevant model code:
```

export class GameTestModel  {
    clientModels = observable(new Array<ILobbyModel>())

    constructor(numberOfClients: number)
    {
        for(let i = 0; i < numberOfClients; i++)  {
            this.clientModels.push(this.makeClientModel(`client${i}`))
        }
    }

    private makeClientModel(clientName: string)
    {
        const newLobby = new LobbyModel();
        newLobby.playerName = clientName;
        return newLobby;
    }
}
```

**Comment by FredyC** [comment]
Sorry, but PR is not a place for discussing your problem. Open a new issue with reproduction, thanks.

**Pull request merged**